realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1051,7 +1051,11 @@ std::unique_ptr<Query> handoverQueryToWorker(jlong bgSharedGroupPtr, jlong query
     if (sg->get_transact_stage() != SharedGroup::transact_Reading) {
         // if the SharedGroup is not in Read Transaction, we position it at the same version as the handover
         sg->begin_read(handoverQuery->version);
-    } else if (sg->get_version_of_current_transaction() != handoverQuery->version) {
+    } else if (sg->get_version_of_current_transaction() < handoverQuery->version) {
+        // going forward
+        LangBindHelper::advance_read(*sg, handoverQuery->version);
+    } else if (handoverQuery->version < sg->get_version_of_current_transaction()) {
+        // going backward
```

Comment:
we do need to go backward, this is most of the time what happens, when you receive a `REALM_CHANGE` an you try to update your query, obviously the UI thread delayed the SharedGroup advance, until the async queries updates. so the handed over query will most likely be behind the latest worker thread shared group